src/openzaak/js/components/admin/external-services/auth-type.js

```diff
@@ -0,0 +1,98 @@
+import React, {useContext, useState} from "react";
+import {SelectInput} from "./select";
+import {TextInput} from "../../../forms/inputs";
+import {ConstantsContext} from "./context";
+
+
+function AuthType(props) {
+    const { index, data } = props;
+    const { values, errors } = data;
+
+    const { authTypeChoices } = useContext(ConstantsContext);
+    const [ selectedAuthType, setSelectedAuthType ] = useState(values.auth_type);
+
+    const id_prefix = (field) => `id_form-${index}-${field}`;
+    const name_prefix = (field) => `form-${index}-${field}`;
+
+
+    const AuthField = (props) => {
```

Comment:
this caught me a bit off-guard in terms of readability - defining another component as a closure. I'd move this to the module level rather. Though I see you need the foo_prefix functions... :thinking: